src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp

```diff
@@ -161,6 +162,12 @@ void JSONEachRowRowInputFormat::readJSONObject(MutableColumns & columns)
     for (size_t key_index = 0; advanceToNextKey(key_index); ++key_index)
     {
         StringRef name_ref = readColumnName(*in);
+        if (seen_columns_count >= total_columns)
```

Comment:
We would skip the duplicated fields check in `readField` in case we read all columns. Considering that I think we should introduce a new setting to enable that.

Reply:
done